BellatrixMergeTransitionAT failures

### Description
Getting quite a few failures from `BellatrixMergeTransitionAcceptanceTest`. e.g:

https://app.circleci.com/pipelines/github/ConsenSys/teku/17338/workflows/6c74ad15-a7af-42ea-932c-1ffd41c3675d/jobs/117774

It looks like it managed to create invalid payloads which shouldn't be possible. There's also a bunch of cases where it failed to get a payloadId.

```
2022-04-03 21:36:02.014 ERROR - Validator   *** Failed to produce block  Slot: 23 Validator: b96ef61
tech.pegasys.teku.spec.logic.common.statetransition.exceptions.StateTransitionException: tech.pegasys.teku.spec.logic.common.statetransition.exceptions.BlockProcessingException: Execution payload parent hash does not match previous execution payload header
at tech.pegasys.teku.spec.logic.common.util.BlockProposalUtil.createNewUnsignedBlock(BlockProposalUtil.java:98) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.spec.Spec.createNewUnsignedBlock(Spec.java:502) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.validator.coordinator.BlockFactory.createUnsignedBlock(BlockFactory.java:54) ~[teku-beacon-validator-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.validator.coordinator.ValidatorApiHandler.createBlock(ValidatorApiHandler.java:312) ~[teku-beacon-validator-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.validator.coordinator.ValidatorApiHandler.lambda$createUnsignedBlock$11(ValidatorApiHandler.java:289) ~[teku-beacon-validator-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.infrastructure.async.SafeFuture.lambda$thenApplyChecked$29(SafeFuture.java:422) ~[teku-infrastructure-async-develop.jar:22.3.2+48-g163c5da]
at java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
at java.util.concurrent.CompletableFuture.complete(Unknown Source) ~[?:?]
at tech.pegasys.teku.dataproviders.generators.CachingTaskQueue.lambda$completePendingTask$9(CachingTaskQueue.java:199) ~[teku-ethereum-dataproviders-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.infrastructure.async.SafeFuture.fromRunnable(SafeFuture.java:154) ~[teku-infrastructure-async-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.infrastructure.async.AsyncRunner.lambda$runAsync$0(AsyncRunner.java:23) ~[teku-infrastructure-async-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.infrastructure.async.SafeFuture.of(SafeFuture.java:73) ~[teku-infrastructure-async-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.infrastructure.async.ScheduledExecutorAsyncRunner.lambda$createRunnableForAction$1(ScheduledExecutorAsyncRunner.java:119) ~[teku-infrastructure-async-develop.jar:22.3.2+48-g163c5da]
at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
at java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: tech.pegasys.teku.spec.logic.common.statetransition.exceptions.BlockProcessingException: Execution payload parent hash does not match previous execution payload header
at tech.pegasys.teku.spec.logic.versions.bellatrix.block.BlockProcessorBellatrix.processExecutionPayload(BlockProcessorBellatrix.java:102) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.spec.logic.versions.bellatrix.block.BlockProcessorBellatrix.processBlock(BlockProcessorBellatrix.java:85) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.spec.logic.common.block.AbstractBlockProcessor.lambda$processUnsignedBlock$7(AbstractBlockProcessor.java:299) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.spec.datastructures.state.beaconstate.common.AbstractBeaconState.updated(AbstractBeaconState.java:64) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.spec.logic.common.block.AbstractBlockProcessor.processUnsignedBlock(AbstractBlockProcessor.java:297) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
at tech.pegasys.teku.spec.logic.common.util.BlockProposalUtil.createNewUnsignedBlock(BlockProposalUtil.java:86) ~[teku-ethereum-spec-develop.jar:22.3.2+48-g163c5da]
... 16 more
```

--- a/acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/BellatrixMergeTransitionAcceptanceTest.java
+++ b/acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/BellatrixMergeTransitionAcceptanceTest.java
@@ -14,7 +14,6 @@
 package tech.pegasys.teku.test.acceptance;
 
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 import tech.pegasys.teku.infrastructure.time.SystemTimeProvider;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
@@ -23,7 +22,6 @@ import tech.pegasys.teku.test.acceptance.dsl.BesuNode;
 import tech.pegasys.teku.test.acceptance.dsl.TekuNode;
 import tech.pegasys.teku.test.acceptance.dsl.tools.deposits.ValidatorKeystores;
 
-@Disabled("While investigating flakiness")
 public class BellatrixMergeTransitionAcceptanceTest extends AcceptanceTestBase {
   private static final String NETWORK_NAME = "less-swift";
 

--- a/eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/forkchoice/ForkChoiceTestExecutor.java
+++ b/eth-reference-tests/src/referenceTest/java/tech/pegasys/teku/reference/phase0/forkchoice/ForkChoiceTestExecutor.java
@@ -180,7 +180,9 @@ public class ForkChoiceTestExecutor implements TestExecutor {
                       .readFixedBytes(Bytes32.SIZE)
                       .toUnsignedBigInteger(ByteOrder.LITTLE_ENDIAN));
           reader.readFixedBytes(Bytes32.SIZE); // Read difficulty even though we don't use it.
-          return new PowBlock(blockHash, parentHash, totalDifficulty);
+          // We don't get a timestamp but as long as it's in the past that's fine
+          final UInt64 timestamp = UInt64.ZERO;
+          return new PowBlock(blockHash, parentHash, totalDifficulty, timestamp);
         });
   }
 

--- a/ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/client/Web3JExecutionEngineClient.java
+++ b/ethereum/executionlayer/src/main/java/tech/pegasys/teku/ethereum/executionlayer/client/Web3JExecutionEngineClient.java
@@ -38,6 +38,7 @@ import tech.pegasys.teku.ethereum.executionlayer.client.serialization.SignedBeac
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.infrastructure.bytes.Bytes8;
 import tech.pegasys.teku.infrastructure.time.TimeProvider;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
 import tech.pegasys.teku.spec.datastructures.execution.PowBlock;
 
@@ -94,7 +95,8 @@ public class Web3JExecutionEngineClient implements ExecutionEngineClient {
         : new PowBlock(
             Bytes32.fromHexStringStrict(eth1Block.getHash()),
             Bytes32.fromHexStringStrict(eth1Block.getParentHash()),
-            UInt256.valueOf(eth1Block.getTotalDifficulty()));
+            UInt256.valueOf(eth1Block.getTotalDifficulty()),
+            UInt64.valueOf(eth1Block.getTimestamp()));
   }
 
   @Override

--- a/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/PowBlock.java
+++ b/ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/execution/PowBlock.java
@@ -16,17 +16,21 @@ package tech.pegasys.teku.spec.datastructures.execution;
 import com.google.common.base.MoreObjects;
 import org.apache.tuweni.bytes.Bytes32;
 import org.apache.tuweni.units.bigints.UInt256;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 
 public class PowBlock {
 
   private final Bytes32 blockHash;
   private final Bytes32 parentHash;
   private final UInt256 totalDifficulty;
+  private final UInt64 blockTimestamp;
 
-  public PowBlock(Bytes32 blockHash, Bytes32 parentHash, UInt256 totalDifficulty) {
+  public PowBlock(
+      Bytes32 blockHash, Bytes32 parentHash, UInt256 totalDifficulty, UInt64 blockTimestamp) {
     this.blockHash = blockHash;
     this.parentHash = parentHash;
     this.totalDifficulty = totalDifficulty;
+    this.blockTimestamp = blockTimestamp;
   }
 
   public Bytes32 getBlockHash() {
@@ -41,12 +45,17 @@ public class PowBlock {
     return totalDifficulty;
   }
 
+  public UInt64 getBlockTimestamp() {
+    return blockTimestamp;
+  }
+
   @Override
   public String toString() {
     return MoreObjects.toStringHelper(this)
         .add("blockHash", blockHash)
         .add("parentHash", parentHash)
         .add("totalDifficulty", totalDifficulty)
+        .add("blockTimestamp", blockTimestamp)
         .toString();
   }
 }

--- a/ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/bellatrix/helpers/BellatrixTransitionHelpersTest.java
+++ b/ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/versions/bellatrix/helpers/BellatrixTransitionHelpersTest.java
@@ -151,7 +151,7 @@ class BellatrixTransitionHelpersTest {
 
   private PowBlock withPowBlock(final Bytes32 hash, final UInt256 totalDifficulty) {
     final PowBlock powBlock =
-        new PowBlock(hash, dataStructureUtil.randomBytes32(), totalDifficulty);
+        new PowBlock(hash, dataStructureUtil.randomBytes32(), totalDifficulty, UInt64.ZERO);
     when(executionEngine.getPowBlock(powBlock.getBlockHash()))
         .thenReturn(completedFuture(Optional.of(powBlock)));
     return powBlock;

--- a/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
+++ b/ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitor.java
@@ -24,6 +24,7 @@ import tech.pegasys.teku.infrastructure.async.Cancellable;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.infrastructure.collections.TekuPair;
 import tech.pegasys.teku.infrastructure.logging.EventLogger;
+import tech.pegasys.teku.infrastructure.time.TimeProvider;
 import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.SpecMilestone;
@@ -40,6 +41,7 @@ public class TerminalPowBlockMonitor {
   private static final Logger LOG = LogManager.getLogger();
 
   private final EventLogger eventLogger;
+  private final TimeProvider timeProvider;
   private final ExecutionEngineChannel executionEngine;
   private final AsyncRunner asyncRunner;
   private Optional<Cancellable> timer = Optional.empty();
@@ -60,13 +62,15 @@ public class TerminalPowBlockMonitor {
       final RecentChainData recentChainData,
       final ForkChoiceNotifier forkChoiceNotifier,
       final AsyncRunner asyncRunner,
-      final EventLogger eventLogger) {
+      final EventLogger eventLogger,
+      final TimeProvider timeProvider) {
     this.executionEngine = executionEngine;
     this.asyncRunner = asyncRunner;
     this.spec = spec;
     this.recentChainData = recentChainData;
     this.forkChoiceNotifier = forkChoiceNotifier;
     this.eventLogger = eventLogger;
+    this.timeProvider = timeProvider;
   }
 
   public synchronized void start() {
@@ -230,6 +234,10 @@ public class TerminalPowBlockMonitor {
               final UInt256 totalDifficulty = powBlock.getTotalDifficulty();
               if (totalDifficulty.compareTo(specConfigBellatrix.getTerminalTotalDifficulty()) < 0) {
                 LOG.trace("checkTerminalBlockByTTD: Total Terminal Difficulty not reached.");
+                return SafeFuture.COMPLETE;
+              }
+              if (powBlock.getBlockTimestamp().isGreaterThan(timeProvider.getTimeInSeconds())) {
+                LOG.trace("checkTerminalBlockByTTD: Chain head is in the future, ignoring for now");
                 return SafeFuture.COMPLETE;
               }
 

--- a/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java
+++ b/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceTest.java
@@ -801,12 +801,14 @@ class ForkChoiceTest {
     final Bytes32 terminalBlockHash = dataStructureUtil.randomBytes32();
     final Bytes32 terminalBlockParentHash = dataStructureUtil.randomBytes32();
     final PowBlock terminalBlock =
-        new PowBlock(terminalBlockHash, terminalBlockParentHash, terminalTotalDifficulty.plus(1));
+        new PowBlock(
+            terminalBlockHash, terminalBlockParentHash, terminalTotalDifficulty.plus(1), ZERO);
     final PowBlock terminalParentBlock =
         new PowBlock(
             terminalBlockParentHash,
             dataStructureUtil.randomBytes32(),
-            terminalTotalDifficulty.subtract(1));
+            terminalTotalDifficulty.subtract(1),
+            ZERO);
     executionEngine.addPowBlock(terminalBlock);
     executionEngine.addPowBlock(terminalParentBlock);
     final SignedBlockAndState epoch4Block =

--- a/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionBlockValidatorTest.java
+++ b/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/MergeTransitionBlockValidatorTest.java
@@ -25,6 +25,7 @@ import org.junit.jupiter.api.Test;
 import tech.pegasys.teku.core.ChainBuilder.BlockOptions;
 import tech.pegasys.teku.infrastructure.async.SafeFuture;
 import tech.pegasys.teku.infrastructure.exceptions.FatalServiceFailureException;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.spec.Spec;
 import tech.pegasys.teku.spec.TestSpecFactory;
 import tech.pegasys.teku.spec.datastructures.blocks.SignedBlockAndState;
@@ -252,10 +253,14 @@ class MergeTransitionBlockValidatorTest {
         new PowBlock(
             getExecutionPayload(transitionBlock).getParentHash(),
             terminalBlockParentHash,
-            terminalBlockDifficulty));
+            terminalBlockDifficulty,
+            UInt64.ZERO));
     executionEngine.addPowBlock(
         new PowBlock(
-            terminalBlockParentHash, dataStructureUtil.randomBytes32(), ttdBlockParentDifficulty));
+            terminalBlockParentHash,
+            dataStructureUtil.randomBytes32(),
+            ttdBlockParentDifficulty,
+            UInt64.ZERO));
   }
 
   private SignedBlockAndState generateNonfinalizedTransition() {

--- a/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitorTest.java
+++ b/ethereum/statetransition/src/test/java/tech/pegasys/teku/statetransition/forkchoice/TerminalPowBlockMonitorTest.java
@@ -17,6 +17,7 @@ import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.atLeastOnce;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
@@ -56,6 +57,7 @@ public class TerminalPowBlockMonitorTest {
   private static final UInt256 TTD = UInt256.valueOf(10_000_000);
   private static final Bytes32 TERMINAL_BLOCK_HASH = Bytes32.random();
   private static final UInt64 TERMINAL_BLOCK_EPOCH = UInt64.valueOf(2);
+  private static final UInt64 TIME_IN_PAST = UInt64.valueOf(123);
 
   private final ExecutionEngineChannel executionEngine = mock(ExecutionEngineChannel.class);
   private final StubTimeProvider timeProvider = StubTimeProvider.withTimeInSeconds(10_000);
@@ -131,7 +133,13 @@ public class TerminalPowBlockMonitorTest {
 
     terminalPowBlockMonitor =
         new TerminalPowBlockMonitor(
-            executionEngine, spec, recentChainData, forkChoiceNotifier, asyncRunner, eventLogger);
+            executionEngine,
+            spec,
+            recentChainData,
+            forkChoiceNotifier,
+            asyncRunner,
+            eventLogger,
+            timeProvider);
 
     terminalPowBlockMonitor.onNodeSyncStateChanged(true);
   }
@@ -167,7 +175,8 @@ public class TerminalPowBlockMonitorTest {
     final Bytes32 headBlockHash = dataStructureUtil.randomBytes32();
     final Bytes32 headBlockParentHash = dataStructureUtil.randomBytes32();
     when(executionEngine.getPowChainHead())
-        .thenReturn(completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD)));
+        .thenReturn(
+            completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD, TIME_IN_PAST)));
     when(executionEngine.getPowBlock(headBlockParentHash))
         .thenReturn(
             completedFuture(
@@ -175,7 +184,8 @@ public class TerminalPowBlockMonitorTest {
                     new PowBlock(
                         headBlockParentHash,
                         dataStructureUtil.randomBytes32(),
-                        TTD.subtract(10)))));
+                        TTD.subtract(10),
+                        TIME_IN_PAST))));
 
     terminalPowBlockMonitor.start();
 
@@ -213,7 +223,11 @@ public class TerminalPowBlockMonitorTest {
     when(executionEngine.getPowChainHead())
         .thenReturn(
             completedFuture(
-                new PowBlock(headBlockHash, dataStructureUtil.randomBytes32(), TTD.subtract(10))));
+                new PowBlock(
+                    headBlockHash,
+                    dataStructureUtil.randomBytes32(),
+                    TTD.subtract(10),
+                    TIME_IN_PAST)));
 
     asyncRunner.executeQueuedActions();
 
@@ -224,7 +238,8 @@ public class TerminalPowBlockMonitorTest {
     headBlockHash = dataStructureUtil.randomBytes32();
     headBlockParentHash = dataStructureUtil.randomBytes32();
     when(executionEngine.getPowChainHead())
-        .thenReturn(completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD)));
+        .thenReturn(
+            completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD, TIME_IN_PAST)));
     when(executionEngine.getPowBlock(headBlockParentHash))
         .thenReturn(
             completedFuture(
@@ -232,7 +247,8 @@ public class TerminalPowBlockMonitorTest {
                     new PowBlock(
                         headBlockParentHash,
                         dataStructureUtil.randomBytes32(),
-                        TTD.subtract(10)))));
+                        TTD.subtract(10),
+                        TIME_IN_PAST))));
 
     asyncRunner.executeQueuedActions();
 
@@ -253,12 +269,18 @@ public class TerminalPowBlockMonitorTest {
     headBlockHash = dataStructureUtil.randomBytes32();
     headBlockParentHash = dataStructureUtil.randomBytes32();
     when(executionEngine.getPowChainHead())
-        .thenReturn(completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD.add(10))));
+        .thenReturn(
+            completedFuture(
+                new PowBlock(headBlockHash, headBlockParentHash, TTD.add(10), TIME_IN_PAST)));
     when(executionEngine.getPowBlock(headBlockParentHash))
         .thenReturn(
             completedFuture(
                 Optional.of(
-                    new PowBlock(headBlockParentHash, dataStructureUtil.randomBytes32(), TTD))));
+                    new PowBlock(
+                        headBlockParentHash,
+                        dataStructureUtil.randomBytes32(),
+                        TTD,
+                        TIME_IN_PAST))));
 
     asyncRunner.executeQueuedActions();
 
@@ -270,7 +292,9 @@ public class TerminalPowBlockMonitorTest {
     headBlockHash = dataStructureUtil.randomBytes32();
     headBlockParentHash = dataStructureUtil.randomBytes32();
     when(executionEngine.getPowChainHead())
-        .thenReturn(completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD.add(10))));
+        .thenReturn(
+            completedFuture(
+                new PowBlock(headBlockHash, headBlockParentHash, TTD.add(10), TIME_IN_PAST)));
     when(executionEngine.getPowBlock(headBlockParentHash))
         .thenReturn(
             completedFuture(
@@ -278,7 +302,8 @@ public class TerminalPowBlockMonitorTest {
                     new PowBlock(
                         headBlockParentHash,
                         dataStructureUtil.randomBytes32(),
-                        TTD.subtract(10)))));
+                        TTD.subtract(10),
+                        TIME_IN_PAST))));
 
     asyncRunner.executeQueuedActions();
 
@@ -299,6 +324,39 @@ public class TerminalPowBlockMonitorTest {
         .exchangeTransitionConfiguration(localTransitionConfiguration);
     verifyNoMoreInteractions(executionEngine);
     verifyNoMoreInteractions(eventLogger);
+  }
+
+  @Test
+  public void shouldNotSelectTTDBlockWithTimestampInFuture() {
+    Bytes32 headBlockHash;
+    Bytes32 headBlockParentHash;
+
+    setUpTTDConfig();
+
+    terminalPowBlockMonitor.start();
+
+    // AT BELLATRIX FORK, TTD reached but block in future - should not notify
+    goToSlot(BELLATRIX_FORK_EPOCH.times(spec.getGenesisSpecConfig().getSlotsPerEpoch()));
+    headBlockHash = dataStructureUtil.randomBytes32();
+    headBlockParentHash = dataStructureUtil.randomBytes32();
+    final UInt64 timeInFuture = timeProvider.getTimeInSeconds().plus(1);
+    when(executionEngine.getPowChainHead())
+        .thenReturn(
+            completedFuture(new PowBlock(headBlockHash, headBlockParentHash, TTD, timeInFuture)));
+    when(executionEngine.getPowBlock(headBlockParentHash))
+        .thenReturn(
+            completedFuture(
+                Optional.of(
+                    new PowBlock(
+                        headBlockParentHash,
+                        dataStructureUtil.randomBytes32(),
+                        TTD.subtract(10),
+                        TIME_IN_PAST))));
+
+    asyncRunner.executeQueuedActions();
+
+    verify(eventLogger, never()).terminalPowBlockDetected(headBlockHash);
+    verify(forkChoiceNotifier, never()).onTerminalBlockReached(headBlockHash);
   }
 
   @Test
@@ -344,7 +402,10 @@ public class TerminalPowBlockMonitorTest {
             completedFuture(
                 Optional.of(
                     new PowBlock(
-                        TERMINAL_BLOCK_HASH, dataStructureUtil.randomBytes32(), UInt256.ONE))));
+                        TERMINAL_BLOCK_HASH,
+                        dataStructureUtil.randomBytes32(),
+                        UInt256.ONE,
+                        TIME_IN_PAST))));
 
     asyncRunner.executeQueuedActions();
 

--- a/services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java
+++ b/services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java
@@ -381,7 +381,8 @@ public class BeaconChainController extends Service implements BeaconChainControl
                   recentChainData,
                   forkChoiceNotifier,
                   beaconAsyncRunner,
-                  EVENT_LOG));
+                  EVENT_LOG,
+                  timeProvider));
     }
   }
 

